### DISCUSSION TITLE
How to call the data loader inside pl.LightningModule?

### DISCUSSION THREAD

**Discussion by KinWaiCheuk** [Lightning-AI/pytorch-lightning#8973] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
My scheduler takes the total number of steps `len(train_loader)*epochs` as one of its arguments.
I know inside the pl.LightningModule, I can get the number of epochs via `self.trainer.max_epochs`.

But how to get the data_loader?
I have tried `self.trainer.train_dataloaders`, `self.train_dataloaders`, but it does not return me the data loader I want.
